AnnService/inc/Core/SearchQuery.h

```diff
@@ -230,9 +187,7 @@ class QueryResult
 
     bool m_withMeta;
 
-    std::unique_ptr<BasicResult[]> m_results;
-
-    std::unique_ptr<ByteArray[]> m_metadatas;
+    std::shared_ptr<BasicResult> m_results;
```

Comment:
if you do need a shared_array, maybe it is better to convert ByteArray into a template